crates/pallet-domains/src/staking.rs

```diff
@@ -147,6 +149,41 @@ pub(crate) fn do_register_operator<T: Config>(
     })
 }
 
+pub(crate) fn do_nominate_operator<T: Config>(
+    operator_id: OperatorId,
+    nominator_id: T::AccountId,
+    amount: BalanceOf<T>,
+) -> Result<(), Error> {
+    OperatorPools::<T>::try_mutate(operator_id, |maybe_operator| {
+        let operator = maybe_operator.as_mut().ok_or(Error::UnknownOperator)?;
+
+        // reserve stake balance
+        ensure!(
+            amount >= operator.minimum_nominator_stake,
+            Error::MinimumNominatorStake
+        );
+
+        ensure!(
+            T::Currency::balance_freezable(&nominator_id) >= amount,
+            Error::InsufficientBalance
+        );
+
+        T::Currency::set_freeze(
```

Comment:
Not sure if I understand correctly, seems `set_freeze` only ensures the user account balance is not below the frozen amount which means a nominator can nominate multiple operators as long as it has `max_of_all(minimum_nominator_stake)` amount of balance instead of `sum_of_all(minimum_nominator_stake)`.

Reply:
Seems resolved by `freeze_account_balance_to_operator`